src/util.rs

```diff
@@ -106,14 +106,17 @@ impl<I: invariant::Validity> ValidityVariance<I> for Invariant {
 
 // SAFETY:
 // - Per [1], `MaybeUninit<T>` has the same size as `T`.
-// - Per [1], `MaybeUninit<T>` has `UnsafeCell`s at the same byte ranges as
-//   `Inner = T`, and `UnsafeCell`s at the same byte offsets as `T`.
 // - See inline comments for other safety justifications.
 //
 // [1] TODO(#896): Write a safety proof before the next stable release.
 unsafe impl<T, I: Invariants> TransparentWrapper<I> for MaybeUninit<T> {
     type Inner = T;
 
+    // SAFETY: Per [1], `MaybeUninit<T>` has `UnsafeCell`s at the same byte
+    // ranges as `Inner = T`, and `UnsafeCell`s at the same byte offsets as `T`.
```

Comment:
What's the distinction you're drawing here between ranges and offsets?

Reply:
I should perhaps say "covering" the same byte ranges. E.g., the type `#[repr(C)] struct Foo(UnsafeCell<()>, usize)` doesn't have any byte ranges which are covered by `UnsafeCell`s, but there is a ZST `UnsafeCell` at byte offset 0. Until https://github.com/rust-lang/unsafe-code-guidelines/issues/495 is resolved (and until we remove the restriction from `NoCell` that even ZST `UnsafeCell`s are banned), this distinction is required for soundness.